Framework/test/testQualitySender.cxx

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    testQualitySender.cxx
+/// \author  Michal Tichak
+///
+
+#include <DataSampling/DataSampling.h>
+#include "QualityControl/InfrastructureGenerator.h"
+#include "QualityControl/QualitySender.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::utilities;
+
+void customize(std::vector<CompletionPolicy>& policies)
+{
+  DataSampling::CustomizeInfrastructure(policies);
+  quality_control::customizeInfrastructure(policies);
+}
+
+#include <Framework/runDataProcessing.h>
+#include <Framework/ControlService.h>
+#include <Configuration/ConfigurationFactory.h>
+#include <Configuration/ConfigurationInterface.h>
+#include <TH1F.h>
+#include <QualityControl/QualitySender.h>
+#include <QualityControl/QualityObject.h>
+
+using namespace o2::configuration;
+
+WorkflowSpec defineDataProcessing(ConfigContext const&)
+{
+  WorkflowSpec specs;
+
+  quality_control::flag_sender::QualitySender sender("grpcUri");
```

Comment:
For the sake of more precise testing later, you could consider `QualitySender` constructor to take a callback to send data to Bookkeeping, while for testing you could put there some testing logic instead, which could print out e.g. a fatal log if data is wrong.